homeassistant/components/homekit/__init__.py

```diff
@@ -92,6 +92,10 @@ def get_accessory(hass, state, aid, config):
             return TYPES['HumiditySensor'](hass, state.entity_id, state.name,
                                            aid=aid)
 
+    elif state.domain == 'binary_sensor' or state.domain == 'device_tracker':
+        _LOGGER.debug('Add "%s" as "%s"', state.entity_id, 'BinarySensor')
+        return TYPES['BinarySensor'](hass, state, aid=aid)
```

Comment:
You don't need to pass in the state, only the entity_id: `state.entity_id`.

Reply:
Only the entity_id is not enough, we need state.attributes to check binary sensor type. I'll follow your style to pass name, entity_id and one more param state.attributes.